src/databricks/labs/remorph/reconcile/recon_config.py

```diff
@@ -307,3 +309,73 @@ class ReconcileOutput:
 class ReconcileRecordCount:
     source: int = 0
     target: int = 0
+
+
+@dataclass
+class Aggregate:
+    agg_columns: list[str]
+    type: str
+    group_by_columns: list[str] | None = None
+
+    def __post_init__(self):
+        self.agg_columns = to_lower_case(self.agg_columns)
+        self.type = self.type.lower()
+        self.group_by_columns = to_lower_case(self.group_by_columns) if self.group_by_columns else None
+
+    def get_agg_type(self):
+        return self.type
+
+    @property
+    def group_by_columns_as_str(self):
```

Comment:
I see so many list to string function create a function generic  to tackle this.

Reply:
Same as here https://github.com/databrickslabs/remorph/pull/740#discussion_r1696937786

Reply:
what I meant was I see 2 properties`group_by_columns_as_str` and `agg_columns_as_str) both convert `list[str]` to `str`, I was recommending those to be handled in single place.

Reply:
Done, added the common function `_join_columns` to handle this.